File: 3k-1s/architecture/laba4/benchit/kernel/utilities/skeleton/F77/0/0/intervallist/work.f
CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC
C BenchIT - Performance Measurement for Scientific Applications
C Contact: developer@benchit.org
C
C $Id: simple.f 1 2009-09-11 12:26:19Z william $
C $URL: svn+ssh://william@rupert.zih.tu-dresden.de/svn-base/benchit-root/BenchITv6/kernel/utilities/skeleton/F77/0/0/simple/simple.f $
C For license details see COPYING in the package base directory
CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC
C Kernel: Fortran-Skeleton
CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC

	SUBROUTINE work_1(n)
	INTEGER*4 n
	do 100 i=1,n
	   WRITE (*,*) "Hello world1!"
 100	CONTINUE
	RETURN
	END

	SUBROUTINE work_2(n)
	INTEGER*4 n
	do 100 i=1,n
	   WRITE (*,*) "Hello world2!"
 100	CONTINUE
	RETURN
	END

File: 3k-1s/architecture/laba4/benchit/kernel/numerical/gemv/C/0/0/float/work.c
/********************************************************************
 * BenchIT - Performance Measurement for Scientific Applications
 * Contact: developer@benchit.org
 *
 * $Id: work.c 1 2009-09-11 12:26:19Z william $
 * $URL: svn+ssh://william@rupert.zih.tu-dresden.de/svn-base/benchit-root/BenchITv6/kernel/numerical/gemv/C/0/0/float/work.c $
 * For license details see COPYING in the package base directory
 *******************************************************************/
/* Kernel: C SGEMV kernel
 *******************************************************************/

#include "work.h"

void ij_(int sizeVector,int sizeAusgabe,float alpha,float beta, float *x, float *A, float *y)
{
	int i,j;
        float temp = 0.0;
	for (j=0;j<sizeAusgabe;j++)
	{
		y[j]=beta*y[j];
	}
	//
	// now : x=x, A=A, y=beta*y
	//

	for (i=0;i<sizeVector;i++)
	{
                temp=alpha*x[i];
		for (j=0;j<sizeAusgabe;j++)
		{
			y[j]=y[j]+A[i*sizeAusgabe+j]*temp;
		}
	}
}

void ji_(int sizeVector,int sizeAusgabe,float alpha,float beta, float *x, float *A, float *y)
{
	int i,j;
        float temp = 0.0;
	for (j=0;j<sizeAusgabe;j++)
	{
		y[j]=beta*y[j];
	}
	//
	// now : x=x, A=A, y=beta*y
	//

	for (j=0;j<sizeAusgabe;j++)
	{
                temp=0.0;
		for (i=0;i<sizeVector;i++)
		{
			temp=temp+A[i*sizeAusgabe+j]*x[i];
		}
                temp=temp*alpha;
                y[j]=y[j]+temp;
	}
}

